test/board_test.rb

```diff
@@ -54,4 +54,26 @@ def test_it_will_deny_diagonals
     refute @board.valid_placement?(@submarine, ["C2", "D3"])
     refute @board.valid_placement?(@submarine, ["D2", "E3"])
   end
+
+  def test_it_can_place_ships
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+    cell_1 = @board.cells["A1"]
+    cell_2 = @board.cells["A2"]
+    cell_3 = @board.cells["A3"]
+    cell_4 = @board.cells["A4"]
+
+    assert cell_3.ship == cell_2.ship
+    refute cell_4.ship == cell_3.ship
+  end
+
+  def test_ships_cannot_overlap
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+    cell_1 = @board.cells["A1"]
+    cell_2 = @board.cells["A2"]
+    cell_3 = @board.cells["A3"]
+
+    refute @board.valid_placement?(@submarine, ["A1", "B1"])
+    refute @board.valid_placement?(@submarine, ["A3", "A4"])
+    assert @board.valid_placement?(@submarine, ["A4", "B4"])
```

Comment:
May also be worthwhile testing cells that are not in the hash, for example ["A5", "B5"] because I don't think there's technically any check in the valid_placement? for them.